### DISCUSSION TITLE
How to setup memory for getStaticProps functions

### DISCUSSION THREAD

**Discussion by thiagoenge** [vercel/community#206] [Help] [answered]
Reading the docs I saw that limit a memory usage is a good strategy to decrease billing execution time, once the default is 1gb and the logs shows that the maximum usage is about 300mb, but our serverless functions belongs to exported getStaticProps, we don't use `/api` since our data source is appsync. Because of that the following[^1]   is not working, isn't it right?

```
{
  "functions": {
    "pages/post/index.js": {
      "memory": 512,
    }
  } 
}
```

So the question is, how do we set the amount of memory to 512mb in the function that run on getStaticProps, instead use 1gb default

An example of our folder structure is:
/pages
  - [post].js 
    - getStaticPaths (runs on build, is ok!)
    - getStaticProps (runs on non static generated path, consuming 1gb by default, we wish to reduce to 512mb)

I've already been trying to post this question on NextJs discord community and I didn't get help .

[^1]: vercel.json

**Comment by samcx** [comment, marked as answer]
Unfortunately, it's not possible to modify the memory from the Serverless Function behind your `getStaticProps` (e.g. from ISR, `fallback: blocking`).

**Reply by thiagoenge** [reply]
Thanks for answer @samsisle, any chance to get that in the future? Our project uses ISR in all pages, the billed execution time on vercel is increasing to much, even with 30 days revalidate =/

**Reply by Pieparker** [reply]
Feel free to [+1 the feature suggestion](https://github.com/vercel/community/discussions/183)!
